### DISCUSSION TITLE
Upload TFT

### DISCUSSION THREAD

**Discussion by atvecko** [Blackymas/NSPanel_HA_Blueprint#323] [Q&A] [answered]
After 8 hours, 40% is uploaded... I stopped uploading. System data error is displayed. Any idea what to do with it?

**Comment by deejaybeam** [comment]
PLEASE! - READ the three pinned posts in the issues: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues
Read the warning in the in installation-part of the wiki: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-First-Steps---Installation-and-Setup#code
Read the Release notes: https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.1.0

**Comment by atvecko** [comment]
I did everything correctly, I have a functional ESPHome (latest version), integrated and communicating with HA, the name and path to the TFT file are set correctly. After running "Update TFT display" the file started uploading but very, very, very slowly, after 8 hours 40% was uploaded. I have no idea what to do now. I tried to change the path to TFT (from local URL to GitHub) and repeat the upload. It has been running for 2 hours and the display shows "System data ERROR". Is it possible to upload a TFT file manually?

**Comment by deejaybeam** [comment, marked as answer]
I think you don´t have read the pinned issues?
https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/166
https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/177
https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/197

Setting to "external - url" and using "https" is a bad idea, this can make things only worse.
Interupt upload can brick the nextion.
Manual - direct upload of tft is described in #166
